apps/website-25/src/components/courses/exercises/MultipleChoice.tsx

```diff
@@ -1,51 +1,84 @@
 import CTALinkOrButton from '@bluedot/ui/src/CTALinkOrButton';
 import clsx from 'clsx';
-import React from 'react';
+import React, { useCallback, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 
 type MultipleChoiceProps = {
   // Required
-  title: string;
+  answer: string;
   description: string;
+  onExerciseSubmit: (savedExerciseResponse: string, completed?: boolean) => void;
   options: string;
-  answer: string;
+  title: string;
+  // Optional
   className?: string;
+  exerciseResponse?: string;
+};
+
+type FormData = {
+  answer: string;
 };
 
 const MultipleChoice: React.FC<MultipleChoiceProps> = ({
-  title,
-  description,
-  options,
   answer,
   className,
+  description,
+  exerciseResponse,
+  onExerciseSubmit,
+  options,
+  title,
 }) => {
   /**
    * Options are stored as a string with newlines
    * Format them to be an array of strings with no empty strings (i.e., removing trailing return statements)
    */
   const formattedOptions = (options.split('\n').map((o) => o.trim()).filter((o) => o !== ''));
   const formattedAnswer = answer.trim();
+  const formattedExerciseResponse = exerciseResponse?.trim();
 
-  const [selectedOption, setSelectedOption] = React.useState<string | null>(null);
-  const [submittedOption, setSubmittedOption] = React.useState<string | null>(null);
+  const [selectedOption, setSelectedOption] = React.useState<string | null>();
+  const [isEditing, setIsEditing] = React.useState<boolean>(false);
+  const { register, handleSubmit, setValue } = useForm<FormData>({
+    defaultValues: {
+      answer: formattedExerciseResponse || '',
+    },
+  });
 
   const handleOptionSelect = (option: string) => {
     setSelectedOption(option);
-    setSubmittedOption(null); // Reset the submitted option when a new option is selected
+    setIsEditing(true);
   };
 
   const isSelected = (option: string): boolean => {
+    if (!isEditing && formattedExerciseResponse) {
+      return formattedExerciseResponse === option;
+    }
     return selectedOption === option;
   };
 
-  const handleAnswerSubmit = () => {
-    setSubmittedOption(selectedOption);
-  };
+  useEffect(() => {
+    if (formattedExerciseResponse) {
+      setValue('answer', formattedExerciseResponse);
+    }
+  }, [formattedExerciseResponse, setValue]);
+
+  const onSubmit = useCallback(async (data: FormData) => {
+    try {
+      const isAnswerCorrect = data.answer === formattedAnswer;
+      console.log('submitted this answer: ', data.answer, ' and is it correct? ', isAnswerCorrect);
+      await onExerciseSubmit(data.answer, isAnswerCorrect);
+      setIsEditing(false);
+    } catch (e) {
+      console.log('error', e);
+      // Set some state for the error and render it
```

Comment:
nit: so we can find this later

```suggestion
      // TODO: Set some state for the error and render it
```